lang_GENERIC/analyze/AST_to_IL.ml

```diff
@@ -313,6 +348,12 @@ and expr_aux env eorig =
   | G.Call (G.IdSpecial (G.Op op, tok), args) ->
       let args = arguments env args in
       mk_e (Operator ((op, tok), args)) eorig
+  | G.Call (G.IdSpecial ((G.This | G.Super | G.Self | G.Parent), tok) as f, args) ->
```

Comment:
You're right, problem is that before `G.Call(e,args)` we have:
```ocaml
  | G.Call (G.IdSpecial spec, args) ->
      let tok = snd spec in
      let lval = fresh_lval env tok in
      let special = call_special env spec in
```
And `call_special` raises `Impossible` if `spec` is `G.This | G.Super | G.Self | G.Parent)`. But I'll do it in a different way to avoid duplication.